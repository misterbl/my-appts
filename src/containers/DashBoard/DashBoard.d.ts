import { RouteComponentProps, StaticContext } from 'react-router';
import { InjectedIntlProps } from 'react-intl';
import { ActionCreatorsMapObject } from 'redux';
import { IAppState } from '../../types/state';

export interface IDashBoardMapStateToProps {
  user: IAppState['api']['userData'];
}

export type TDashBoard = IDashBoardMapStateToProps &
  InjectedIntlProps &
  RouteComponentProps<any, StaticContext>;
