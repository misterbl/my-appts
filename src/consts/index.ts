import ROUTES from './routes';
import QUERIES from './queries';

export { ROUTES, QUERIES }