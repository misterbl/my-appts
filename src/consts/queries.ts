const queries = (args: any) => ({
  ADD_USER: `mutation {
    addUser (email: "${args.email}", avatar:"${args.avatar}" ) {
      _id
      firstName
      lastName
      avatar
      email
      address
      profileTitle
      profileDescription
     
    }
  }`,
  GET_USER: `mutation {
    getUser (email: "${args.email}") {
      _id
      firstName
      lastName
      avatar
      email
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  GET_USER_BY_ID: `mutation {
    getUserById (_id: "${args.id}") {
      _id
      firstName
      lastName
      avatar
      email
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  GET_USER_INFO: `mutation {
    getUser (email: "${args.email}") {
      _id
      firstName
      lastName
      avatar
      email
      address
    }
  }`,
  GET_AD_INFO: `mutation {
    getUser (email: "${args.email}") {
      _id
      profileTitle
      profileDescription
    }
  }`,
  UPDATE_AD_INFO: `mutation {
    updateUser (_id:"${args._id}" profileTitle: "${
    args.profileTitle
  }", profileDescription: "${args.profileDescription}") {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{name, dob, information, school, gender} 
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  UPDATE_PERSONAL_INFO: `mutation {
    updateUser (_id: "${args._id}", firstName: "${
    args.firstName
  }", lastName: "${args.lastName}", address: "${args.address}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,
  UPDATE_AVATAR: `mutation {
    updateUser (_id: "${args._id}", avatar:"${args.avatar}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,
  UPDATE_LOCATION: `mutation {
    updateUser (_id: "${args._id}", address: "${args.address}", lat: "${
    args.lat
  }", lng: "${args.lng}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,

  UPDATE_ABOUT_YOU: `mutation {
    updateUser (_id: "${args._id}", drivingLicense: ${
    args.drivingLicense
  }, nonSmoker: ${args.nonSmoker}, car:${args.car}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability

    }
  }`,
  UPDATE_AVAILABILITY: `mutation {
    updateUser (_id: "${args._id}", availability: ${JSON.stringify(
    JSON.stringify(args.availability),
  )}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  UPDATE_UNAVAILABILITY: `mutation {
    updateUser (_id: "${args._id}", unavailability: ${JSON.stringify(
    args.unavailability,
  )}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  ADD_CHILD: `mutation {
    updateUser (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
    }
  }`,
  UPDATE_CHILD: `mutation {
    updateChild (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  REMOVE_CHILD: `mutation {
    removeChild (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
    }
  }`,
});

export default queries;
