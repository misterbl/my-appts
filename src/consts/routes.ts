const routes = {
  INDEX: '/',
  CARD: '/card',
  SIGN_IN: '/signin',
  REGISTER: '/register',
  DASHBOARD: './dashboard',
  INBOX: './inbox',
  FAVOURITE: './favourite',
  SEARCH: './search',
};

export default routes;
