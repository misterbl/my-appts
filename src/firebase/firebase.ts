import * as firebase from 'firebase/app';
import 'firebase/auth';


const config = {
    apiKey: "AIzaSyAUFtb8M2Pzfgpn48-F736NVgKlCcc2haQ",
    authDomain: "kidappi-app.firebaseapp.com",
    databaseURL: "https://kidappi-app.firebaseio.com",
    projectId: "kidappi-app",
    storageBucket: "kidappi-app.appspot.com",
    messagingSenderId: "600523020740"
  };

  if (!firebase.apps.length) {
    firebase.initializeApp(config);
  }

  const auth = firebase.auth();
  auth.useDeviceLanguage();
  const provider = new firebase.auth.FacebookAuthProvider()
export {
  auth,
  provider,
};