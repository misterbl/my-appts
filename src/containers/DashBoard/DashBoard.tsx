import * as React from 'react';
import { injectIntl, FormattedMessage } from 'react-intl';
import { withRouter } from 'react-router-dom';
import { TDashBoard, IDashBoardMapStateToProps } from './DashBoard.d';
import Svg from '../../components/Svg';
import {
  AccountIcon,
  chevronRigthIcon,
  signOutIcon,
} from '../../styles/assets';
import { auth } from '../../firebase';
import { IAppState } from '../../types/state';
import { getUserData } from '../../selectors//apiSelectors';
import { connect } from 'react-redux';
import { ROUTES } from '../../consts';

export class DashBoard extends React.Component<TDashBoard> {
  signOut = async () => {
    await auth.doSignOut();
    this.props.history.push(ROUTES.INDEX);
  };
  render() {
    return (
      <>
        {this.props.user && (
          <>
            <div
              onClick={() =>
                // @ts-ignore
                this.props.history.push(`${this.props.user._id}/card`)
              }
              className="flex flex-column pt4 ph5-m"
            >
              u
              <div className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray">
                <div className="flex ml3 justify-between w-100">
                  <span className="flex">
                    <Svg className="mr2" Icon={AccountIcon} />
                    <FormattedMessage id="content|dashboard|profile" />
                  </span>
                  <Svg Icon={chevronRigthIcon} />
                </div>
              </div>
            </div>
            <div
              id="signOut"
              onClick={() => this.signOut()}
              className="flex flex-column pt4 ph5-m"
            >
              <div className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray">
                <div className="flex ml3 justify-between w-100">
                  <span className="flex">
                    <Svg className="mr2" Icon={signOutIcon} />
                    <FormattedMessage id="content|dashboard|signOut" />
                  </span>
                  <Svg Icon={chevronRigthIcon} />
                </div>
              </div>
            </div>
          </>
        )}
      </>
    );
  }
}

export const mapStateToProps = (
  state: IAppState,
): IDashBoardMapStateToProps => ({
  user: getUserData(state),
});

const injectIntlDashBoard = injectIntl(DashBoard);
export default withRouter(
  connect(
    mapStateToProps,
    null,
  )(injectIntlDashBoard),
);
