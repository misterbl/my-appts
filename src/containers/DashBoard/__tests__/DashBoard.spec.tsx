import * as React from 'react';
import { shallow } from 'enzyme';
import historyMock from '../../../testMocks/history.mock';
import { DashBoard } from '../DashBoard';
import IntlMock from '../../../testMocks/intl.mock';
import generateAppState from '../../../testMocks/appState.mock';
import { auth } from '../../../firebase';
import { ROUTES } from '../../../consts';

jest.mock('../../../firebase', () => ({
  auth: {
    doSignOut: jest.fn().mockReturnValue(Promise.resolve({})),
  },
}));

describe('AddInfoForm', () => {
  const props = {
    user: generateAppState().api.userData,
    apiThunk: { updateUser: jest.fn() },
    intl: IntlMock,
    history: historyMock,
    location: historyMock.location,
    match: {
      params: {},
      isExact: true,
      path: '',
      url: '',
    },
  };
  // @ts-ignore
  const wrapper = shallow(<DashBoard {...props} />);
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push() on click', () => {
    const div = wrapper.find('div').at(0);
    div.simulate('click');
    expect(props.history.push).toHaveBeenCalledWith('id/card');
  });
  it('should call auth.doSignOut() on click', async () => {
    const div = wrapper.find('#signOut');
    await div.simulate('click');
    expect(auth.doSignOut).toHaveBeenCalled();
  });
  it('should call props.history.push on click', async () => {
    const div = wrapper.find('#signOut');
    await div.simulate('click');
    expect(props.history.push).toHaveBeenCalledWith(ROUTES.INDEX);
  });
});
