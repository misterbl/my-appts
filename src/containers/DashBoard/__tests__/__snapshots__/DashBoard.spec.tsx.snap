// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddInfoForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashBoard
    apiThunk={
        Object {
            "updateUser": [Function],
          }
    }
    history={
        Object {
            "action": "PUSH",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 123,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "",
              "pathname": "",
              "search": "",
              "state": "",
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    intl={
        Object {
            "defaultFormats": "",
            "defaultLocale": "",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": "",
            "locale": "",
            "messages": Object {},
            "now": [Function],
          }
    }
    location={
        Object {
            "hash": "",
            "key": "",
            "pathname": "",
            "search": "",
            "state": "",
          }
    }
    match={
        Object {
            "isExact": true,
            "params": Object {},
            "path": "",
            "url": "",
          }
    }
    user={
        Object {
            "_id": "id",
            "address": "address",
            "availability": undefined,
            "avatar": "avatar",
            "car": false,
            "children": Array [
              Object {
                "dob": 122322,
                "gender": "boy",
                "id": "childId",
                "information": "childInformation",
                "name": "childName",
                "school": "school",
              },
            ],
            "city": "city",
            "drivingLicense": false,
            "email": "email",
            "firstName": "firstName",
            "lastName": "lastName",
            "lat": 1234,
            "lng": 123445,
            "nonSmoker": false,
            "photoURL": "photU\`rl",
            "postCode": "postCode",
            "profileDescription": "profileDescription",
            "profileTitle": "profileTitle",
            "unavailability": undefined,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Unknown>
        <div
                className="flex flex-column pt4 ph5-m"
                onClick={[Function]}
        >
                u
                <div
                        className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
                >
                        <div
                                className="flex ml3 justify-between w-100"
                        >
                                <span
                                        className="flex"
                                >
                                        <Svg
                                                Icon={[Function]}
                                                className="mr2"
                                        />
                                        <FormattedMessage
                                                id="content|dashboard|profile"
                                                values={Object {}}
                                        />
                                </span>
                                <Svg
                                        Icon={[Function]}
                                />
                        </div>
                </div>
        </div>
        <div
                className="flex flex-column pt4 ph5-m"
                id="signOut"
                onClick={[Function]}
        >
                <div
                        className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
                >
                        <div
                                className="flex ml3 justify-between w-100"
                        >
                                <span
                                        className="flex"
                                >
                                        <Svg
                                                Icon={[Function]}
                                                className="mr2"
                                        />
                                        <FormattedMessage
                                                id="content|dashboard|signOut"
                                                values={Object {}}
                                        />
                                </span>
                                <Svg
                                        Icon={[Function]}
                                />
                        </div>
                </div>
        </div>
</Unknown>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="flex flex-column pt4 ph5-m"
            onClick={[Function]}
>
            u
            <div
                        className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
            >
                        <div
                                    className="flex ml3 justify-between w-100"
                        >
                                    <span
                                                className="flex"
                                    >
                                                <Svg
                                                            Icon={[Function]}
                                                            className="mr2"
                                                />
                                                <FormattedMessage
                                                            id="content|dashboard|profile"
                                                            values={Object {}}
                                                />
                                    </span>
                                    <Svg
                                                Icon={[Function]}
                                    />
                        </div>
            </div>
</div>,
          <div
            className="flex flex-column pt4 ph5-m"
            id="signOut"
            onClick={[Function]}
>
            <div
                        className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
            >
                        <div
                                    className="flex ml3 justify-between w-100"
                        >
                                    <span
                                                className="flex"
                                    >
                                                <Svg
                                                            Icon={[Function]}
                                                            className="mr2"
                                                />
                                                <FormattedMessage
                                                            id="content|dashboard|signOut"
                                                            values={Object {}}
                                                />
                                    </span>
                                    <Svg
                                                Icon={[Function]}
                                    />
                        </div>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "u",
              <div
                className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
>
                <div
                                className="flex ml3 justify-between w-100"
                >
                                <span
                                                className="flex"
                                >
                                                <Svg
                                                                Icon={[Function]}
                                                                className="mr2"
                                                />
                                                <FormattedMessage
                                                                id="content|dashboard|profile"
                                                                values={Object {}}
                                                />
                                </span>
                                <Svg
                                                Icon={[Function]}
                                />
                </div>
</div>,
            ],
            "className": "flex flex-column pt4 ph5-m",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "u",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="flex ml3 justify-between w-100"
>
                  <span
                                    className="flex"
                  >
                                    <Svg
                                                      Icon={[Function]}
                                                      className="mr2"
                                    />
                                    <FormattedMessage
                                                      id="content|dashboard|profile"
                                                      values={Object {}}
                                    />
                  </span>
                  <Svg
                                    Icon={[Function]}
                  />
</div>,
                "className": "h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="flex"
>
                      <Svg
                                            Icon={[Function]}
                                            className="mr2"
                      />
                      <FormattedMessage
                                            id="content|dashboard|profile"
                                            values={Object {}}
                      />
</span>,
                    <Svg
                      Icon={[Function]}
/>,
                  ],
                  "className": "flex ml3 justify-between w-100",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Svg
                          Icon={[Function]}
                          className="mr2"
/>,
                        <FormattedMessage
                          id="content|dashboard|profile"
                          values={Object {}}
/>,
                      ],
                      "className": "flex",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "Icon": [Function],
                          "className": "mr2",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "id": "content|dashboard|profile",
                          "values": Object {},
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "Icon": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
>
              <div
                            className="flex ml3 justify-between w-100"
              >
                            <span
                                          className="flex"
                            >
                                          <Svg
                                                        Icon={[Function]}
                                                        className="mr2"
                                          />
                                          <FormattedMessage
                                                        id="content|dashboard|signOut"
                                                        values={Object {}}
                                          />
                            </span>
                            <Svg
                                          Icon={[Function]}
                            />
              </div>
</div>,
            "className": "flex flex-column pt4 ph5-m",
            "id": "signOut",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="flex ml3 justify-between w-100"
>
                <span
                                className="flex"
                >
                                <Svg
                                                Icon={[Function]}
                                                className="mr2"
                                />
                                <FormattedMessage
                                                id="content|dashboard|signOut"
                                                values={Object {}}
                                />
                </span>
                <Svg
                                Icon={[Function]}
                />
</div>,
              "className": "h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="flex"
>
                    <Svg
                                        Icon={[Function]}
                                        className="mr2"
                    />
                    <FormattedMessage
                                        id="content|dashboard|signOut"
                                        values={Object {}}
                    />
</span>,
                  <Svg
                    Icon={[Function]}
/>,
                ],
                "className": "flex ml3 justify-between w-100",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Svg
                        Icon={[Function]}
                        className="mr2"
/>,
                      <FormattedMessage
                        id="content|dashboard|signOut"
                        values={Object {}}
/>,
                    ],
                    "className": "flex",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "Icon": [Function],
                        "className": "mr2",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "id": "content|dashboard|signOut",
                        "values": Object {},
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "Icon": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Unknown>
          <div
                    className="flex flex-column pt4 ph5-m"
                    onClick={[Function]}
          >
                    u
                    <div
                              className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
                    >
                              <div
                                        className="flex ml3 justify-between w-100"
                              >
                                        <span
                                                  className="flex"
                                        >
                                                  <Svg
                                                            Icon={[Function]}
                                                            className="mr2"
                                                  />
                                                  <FormattedMessage
                                                            id="content|dashboard|profile"
                                                            values={Object {}}
                                                  />
                                        </span>
                                        <Svg
                                                  Icon={[Function]}
                                        />
                              </div>
                    </div>
          </div>
          <div
                    className="flex flex-column pt4 ph5-m"
                    id="signOut"
                    onClick={[Function]}
          >
                    <div
                              className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
                    >
                              <div
                                        className="flex ml3 justify-between w-100"
                              >
                                        <span
                                                  className="flex"
                                        >
                                                  <Svg
                                                            Icon={[Function]}
                                                            className="mr2"
                                                  />
                                                  <FormattedMessage
                                                            id="content|dashboard|signOut"
                                                            values={Object {}}
                                                  />
                                        </span>
                                        <Svg
                                                  Icon={[Function]}
                                        />
                              </div>
                    </div>
          </div>
</Unknown>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="flex flex-column pt4 ph5-m"
              onClick={[Function]}
>
              u
              <div
                            className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
              >
                            <div
                                          className="flex ml3 justify-between w-100"
                            >
                                          <span
                                                        className="flex"
                                          >
                                                        <Svg
                                                                      Icon={[Function]}
                                                                      className="mr2"
                                                        />
                                                        <FormattedMessage
                                                                      id="content|dashboard|profile"
                                                                      values={Object {}}
                                                        />
                                          </span>
                                          <Svg
                                                        Icon={[Function]}
                                          />
                            </div>
              </div>
</div>,
            <div
              className="flex flex-column pt4 ph5-m"
              id="signOut"
              onClick={[Function]}
>
              <div
                            className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
              >
                            <div
                                          className="flex ml3 justify-between w-100"
                            >
                                          <span
                                                        className="flex"
                                          >
                                                        <Svg
                                                                      Icon={[Function]}
                                                                      className="mr2"
                                                        />
                                                        <FormattedMessage
                                                                      id="content|dashboard|signOut"
                                                                      values={Object {}}
                                                        />
                                          </span>
                                          <Svg
                                                        Icon={[Function]}
                                          />
                            </div>
              </div>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "u",
                <div
                  className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
>
                  <div
                                    className="flex ml3 justify-between w-100"
                  >
                                    <span
                                                      className="flex"
                                    >
                                                      <Svg
                                                                        Icon={[Function]}
                                                                        className="mr2"
                                                      />
                                                      <FormattedMessage
                                                                        id="content|dashboard|profile"
                                                                        values={Object {}}
                                                      />
                                    </span>
                                    <Svg
                                                      Icon={[Function]}
                                    />
                  </div>
</div>,
              ],
              "className": "flex flex-column pt4 ph5-m",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "u",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="flex ml3 justify-between w-100"
>
                    <span
                                        className="flex"
                    >
                                        <Svg
                                                            Icon={[Function]}
                                                            className="mr2"
                                        />
                                        <FormattedMessage
                                                            id="content|dashboard|profile"
                                                            values={Object {}}
                                        />
                    </span>
                    <Svg
                                        Icon={[Function]}
                    />
</div>,
                  "className": "h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="flex"
>
                        <Svg
                                                Icon={[Function]}
                                                className="mr2"
                        />
                        <FormattedMessage
                                                id="content|dashboard|profile"
                                                values={Object {}}
                        />
</span>,
                      <Svg
                        Icon={[Function]}
/>,
                    ],
                    "className": "flex ml3 justify-between w-100",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <Svg
                            Icon={[Function]}
                            className="mr2"
/>,
                          <FormattedMessage
                            id="content|dashboard|profile"
                            values={Object {}}
/>,
                        ],
                        "className": "flex",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "Icon": [Function],
                            "className": "mr2",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "id": "content|dashboard|profile",
                            "values": Object {},
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "Icon": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray"
>
                <div
                                className="flex ml3 justify-between w-100"
                >
                                <span
                                                className="flex"
                                >
                                                <Svg
                                                                Icon={[Function]}
                                                                className="mr2"
                                                />
                                                <FormattedMessage
                                                                id="content|dashboard|signOut"
                                                                values={Object {}}
                                                />
                                </span>
                                <Svg
                                                Icon={[Function]}
                                />
                </div>
</div>,
              "className": "flex flex-column pt4 ph5-m",
              "id": "signOut",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="flex ml3 justify-between w-100"
>
                  <span
                                    className="flex"
                  >
                                    <Svg
                                                      Icon={[Function]}
                                                      className="mr2"
                                    />
                                    <FormattedMessage
                                                      id="content|dashboard|signOut"
                                                      values={Object {}}
                                    />
                  </span>
                  <Svg
                                    Icon={[Function]}
                  />
</div>,
                "className": "h3 br2 mt4 flex justify-center items-center shadow-4 pa3 ba b--moon-gray",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="flex"
>
                      <Svg
                                            Icon={[Function]}
                                            className="mr2"
                      />
                      <FormattedMessage
                                            id="content|dashboard|signOut"
                                            values={Object {}}
                      />
</span>,
                    <Svg
                      Icon={[Function]}
/>,
                  ],
                  "className": "flex ml3 justify-between w-100",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Svg
                          Icon={[Function]}
                          className="mr2"
/>,
                        <FormattedMessage
                          id="content|dashboard|signOut"
                          values={Object {}}
/>,
                      ],
                      "className": "flex",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "Icon": [Function],
                          "className": "mr2",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "id": "content|dashboard|signOut",
                          "values": Object {},
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "Icon": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
